### PR TITLE
Update medaka to 1.0.1

### PR DESCRIPTION
**Pull request by stephanflemming**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Website: https://nanoporetech.github.io/medaka/index.html
Github: https://github.com/nanoporetech/medaka
Conda: https://bioconda.github.io/recipes/medaka/README.html
Recipe: https://github.com/bioconda/bioconda-recipes/blob/master/recipes/medaka/meta.yaml
(My) parametero verview: https://docs.google.com/spreadsheets/d/1VkrJnIwMxJMYygBL8BrhtdDfyW1UTbQ8BtbPtr4KnKE/edit?usp=sharing

**Available modules:**

**_Main_**
medaka_consensus
medaka_variants

**_Subcommands_**
medaka compress_bam
medaka consensus
medaka consensus_from_features
medaka fastrle
medaka features
medaka methylation
medaka smolecule
medaka snp
medaka stitch
medaka train
medaka variant

**_Tools subcommands_**
medaka tools annotate
medaka tools classify_variants
medaka tools diploid2haploid
medaka tools diploid2haploid
medaka tools download_models
medaka tools get_alignments_params
medaka tools hdf_to_bed
medaka tools homozygous_regions
medaka tools is_rle_model
medaka tools list_models
medaka tools resolve_models
medaka tools rlebam
medaka tools vcf2tsv

### CODE REVIEW

**Comment by stephanflemming** [comment]
Don't review yet.

**Comment by bernt-matthias** [comment]
Supersedes https://github.com/galaxyproject/tools-iuc/pull/3015?

**Comment by stephanflemming** [comment]
Suggestions? Which of the modules should be added again?

@mmiladi @heylf @bgruening

**Comment by mmiladi** [comment]
`medaka variant` and `medaka consensus` as far as I see. ping @NatPRoach

**Comment by NatPRoach** [comment]
Yes adding `medaka variant` and `medaka consensus` should be sufficient for my purposes. Thanks for all the work on this everyone!

**Comment by mmiladi** [comment]
Thanks for reply @NatPRoach . Do you probably have a minimal small dataset at hand that could be used as integration test case of the Galaxy wrappers?

**Comment by NatPRoach** [comment]
I have a test case somewhere. I'll try to track it down and get it to you.

**Comment by stephanflemming** [comment]
> I have a test case somewhere. I'll try to track it down and get it to you.

Perfect. I wrote you on gitter with an explanation of what I need. Thank you!

File: tools/medaka/macros.xml [hunk 1]
```diff
@@ -0,0 +1,159 @@
+<?xml version="1.0"?>
+<macros>
+    <token name="@TOOL_VERSION@">1.0.1</token>
+    <token name="@WRAPPER_VERSION@">galaxy0</token>
+    <token name="@PROFILE@">19.01</token>
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="@TOOL_VERSION@">medaka</requirement>
+            <requirement type="package" version="1.0.3">pandas</requirement> <!-- should be stored in recipe -->
```
Comment (bgruening) [inline tools/medaka/macros.xml:9]: should be stored in recipe ...

File: tools/medaka/medaka_variant.xml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements"/>
+    <expand macro="version_command"/>
+    <command detect_errors="exit_code"><![CDATA[
+        ## initialize
+        @REF_FASTA@  
+        ## soft link doesn't work      
+        cp '${i}' 'sample.bam' &&
```
Comment (bgruening) [inline tools/medaka/medaka_variant.xml:12]: does a link also work, e.g. ln -s?

Comment (stephanflemming) [reply tools/medaka/medaka_variant.xml:12]: Nope, neither hard nor soft link is accepted.

**Comment by stephanflemming** [comment]
I tried to use the testdata provied [here](https://github.com/nanoporetech/medaka/tree/master/medaka/test/data)

**consensus**
_Test_: Works with `reads_gapped.bam `
_Unclear argument(s)_: `--regions` (what kind of bed file can be provided here?),` --RG`, `--tag_name`, `--tag_value`, `--tag_keep_missing `

**variant**
_Test_: Run with results from `medaka consensus` doesn't work.

**medaka_variant**
_Test_: Run stopps either with `* Failed to call snps from consensus probabilities.` or produces empty (and temporary?) result files like `round_0_hap_mixed_probs.hdf, round_0_hap_mixed_unphased.vcf,  round_1.vcf`
_Unclear argument(s)_: -r

**medaka_consensus**
_Test_: Run stops with `Invalid option: -M.` [See](https://github.com/nanoporetech/medaka/issues/158)

**Comment by cjw85** [comment]
What values can be used for:
`--RG`: this parsed as a `str` and compared to read groups in the input bam.
`--tag_name`: should be a two character `str`
`--tag_value`: parsed to `int`
`--tag_keep_missing`: this is a boolean flag that stores True. It doesn't take a value

What kind of bed file can be used for `--regions`?: there is a simple parser which reads the first three columns as a name:start-stop format.

Unclear argument(s): Does --regions also accept BED files? --- No, but this should be considered a bug. I have raised an issue internally.

How to use `-r` ?  --- whenever medaka refers to "region strings" it is expecting a samtools-style description such as `chr1:3000-6000`

**Comment by stephanflemming** [comment]
Thanks for the reply. I still fail to run
- medaka_variant
- medaka variant

I will write you on gitter.

File: tools/medaka/consensus.xml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        <param argument="--tag_keep_missing" type="boolean" truevalue="--tag_keep_missing" falsevalue="" label="Keep alignments when tag is missing?"/>
+
+        <param name="out" type="select" multiple="true" optional="false" label="Select out file(s)" help="">
+            <option value="result" selected="true">Result</option>
+            <option value="log">Log</option>
+        </param>
+    </inputs>
+
+    <outputs>
+        <data name="out_result" format="data" from_work_dir="result.hdf" label="${tool.name} on ${on_string}: Result">
```
Comment (bgruening) [inline tools/medaka/consensus.xml:82]: format is wrong here ...

File: tools/medaka/medaka_consensus.xml [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            <option value="polished">Polished regions in draft coordinates</option>
+            <option value="log">Log</option>
+        </param>
+    </inputs>
+    <outputs>
+        <!-- standard output -->
+        <data name="out_consensus" format="fasta" from_work_dir="results/consensus.fasta" label="${tool.name} on ${on_string}: Consensus">
+            <filter>'consensus' in out</filter>
+        </data>
+        <data name="out_probs" format="data" from_work_dir="results/consensus_probs.hdf" label="${tool.name} on ${on_string}: Consensus probabilities">
```
Comment (bgruening) [inline tools/medaka/medaka_consensus.xml:44]: there is no format="data"

Comment (nsoranzo) [reply tools/medaka/medaka_consensus.xml:44]: Well, the `data` format exists, it's the most general and unspecified format, and it is used in a number of tools when a proper datatype does not exist in Galaxy yet.

File: tools/medaka/macros.xml (hunk 1 above)
Comment (stephanflemming) [reply tools/medaka/macros.xml]: Just needed by one of the subcommands that will not be included in this wrapper. I removed it.

File: tools/medaka/medaka_consensus.xml (hunk 4 above)
Comment (stephanflemming) [reply tools/medaka/medaka_consensus.xml:44]: Was not sure if the output format is h5, hdf, hdf5, ... and used data as a placeholder. Replaced it now with h5.

**Comment by stephanflemming** [comment]
**consensus**
I used h5 as the output format.
I couldn't test the parameters --RG, --tag_name, --tag_value, and --regions (str and bed file).
Not sure if the help section is sufficient.

**medaka_consensus**
Help section is not sufficient.

**variant**
I couldn't test the parameter --regions.
Testdata causes the error `KeyError: "sequence 'ref' not present"` and needs to be replaced. (See https://github.com/nanoporetech/medaka/issues/159)
Help section is not sufficient.

**medaka_variant**
I couldn't test the parameter -r. The parameter -p causes an error and therefore excluded. (See https://github.com/nanoporetech/medaka/issues/161)
Testdata causes the error '* Failed to call snps from consensus probabilities' not present"' and needs to be replaced. (See https://github.com/nanoporetech/medaka/issues/159)
Help section is not sufficient.

**Comment by nsoranzo** [comment]
A `tools/medaka/tool_data_table_conf.xml.sample` file is missing.

File: tools/medaka/macros.xml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            <option value="r941_prom_snp_g303">r941_prom_snp_g303</option>
+            <option value="r941_prom_snp_g322">r941_prom_snp_g322</option>
+            <option value="r941_prom_variant_g303">r941_prom_variant_g303</option>
+            <option value="r941_prom_variant_g322">r941_prom_variant_g322</option>
+        </param>
+    </xml>
+    <xml name="reference">
+        <conditional name="reference_source">
+            <param name="reference_source_selector" type="select" label="Will you select a reference genome from your history or use a built-in index?" help="Built-ins were indexed using default options. See 'Indexes' section of help below. If you would like to perform self-mapping select 'history' here, then choose your input file as reference.">
+                <option value="cached">Use a built-in genome index</option>
```
Comment (nsoranzo) [inline tools/medaka/macros.xml:75]:
```suggestion
                <option value="cached">Use a built-in genome</option>
```

File: tools/medaka/macros.xml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+                <option value="history">Use a genome from history and build index</option>
```
Comment (nsoranzo) [inline tools/medaka/macros.xml:76]:
```suggestion
                <option value="history">Use a genome from history</option>
```

File: tools/medaka/medaka_consensus.xml [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        #if 'variants' in str($out).split(',')
+            && bgzip -d 'results/variants.vcf.gz'
+        #end if
+    ]]></command>
+    <inputs>
+        <param argument="-i" type="data" format="fastq,fastq.gz,fastqsanger,fastqsanger.gz,fasta,fasta.gz" label="Select basecalls"/>
+        <param argument="-d" type="data" format="fasta,fasta.gz" label="Select assembly" help="The input assembly should be preprocessed with racon."/>
+        <expand macro="model"/>
+        <expand macro="b"/>
+        <param name="out" type="select" multiple="true" optional="false" label="Select output file(s)" help="'Draft To Consensus', 'Variants' and 'Polished regions in draft coordinates' are generated using the parameter -v.">
```
Comment (nsoranzo) [inline tools/medaka/medaka_consensus.xml:32]: We can omit implementation details, I think.

```suggestion
        <param name="out" type="select" multiple="true" optional="false" label="Select output file(s)">
```

Comment (stephanflemming) [reply tools/medaka/medaka_consensus.xml:32]: I like that via `argument=` the underlying parameter is automatically mentioned in brackets as an addition to the content of the help tag. In this case, -v doesn't have a separate input element in this wrapper and is just indirectly available in this select element. I think it's good to mention that.

Comment (stephanflemming) [reply tools/medaka/medaka_consensus.xml:32]: I like that via argument= the underlying parameter is automatically mentioned in brackets as an addition to the content of the help tag. In this case, -v doesn't have a separate input element in this wrapper and is just indirectly available in this select element. I think it's good to mention that.

**Comment by bgruening** [comment]
Thanks @stephanflemming will deploy it now! @nsoranzo thanks for the review!

**Pull request merged**